### ISSUE TITLE
Project budgets and alert thresholds

### ISSUE THREAD

**Issue by carlpett** [terraform-providers/terraform-provider-google#4068]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

We'd like to create billing budgets and corresponding alert thresholds for projects. Currently we need to set these up manually after creating a project with Terraform.

_NOTE: I couldn't find any open GCP APIs to do this currently, but I hope you are in a good position to lobby for opening them_ 

### New or Affected Resource(s)

* google_billing_budget (?)

### Potential Terraform Configuration

```tf
resource "google_project" "my_project" {
  name = "My Project"
  project_id = "your-project-id"
  org_id     = "1234567"
}
resource "google_pubsub_topic" "billing_alerts" {
  name = "billing-alerts"
  project = google_project.my_project.project_id
}

resource "google_billing_budget" "my_budget" {
  name = "My Project Budget"
  project_id = google_project.my_project.project_id
  amount = "LAST_MONTH_SPEND" # Or a specified amount, such as 1500
  include_credits = true
  action {
    percentage = 75
    trigger_on = "actual"
  }
  action {
    percentage = 100
    trigger_on = "forecasted"
  }
  notification {
    type = "pubsub" # If needed for clarity? Currently this is the only supported type
    project_id = google_pubsub_topic.billing_alerts.project
    topic = google_pubsub_topic.billing_alerts.name
  }
}
```

### References
https://cloud.google.com/billing/docs/how-to/budgets

**Comment by emilymye** [comment]
We can certainly ask!

**Comment by shivaprasad201** [comment]
https://issuetracker.google.com/131268009 I guess the gcloud billing apis are still a WIP.

**Comment by tonglil** [comment]
The api was released: https://cloud.google.com/billing/docs/how-to/budget-api-overview

**Comment by danawillow** [comment]
This is now done, and will be released in version 3.1.0: https://github.com/GoogleCloudPlatform/magic-modules/pull/2734
